tests/test_client_functional.py

```diff
@@ -2202,6 +2202,30 @@ async def handler_redirect(request):
     assert data == body
 
 
+@pytest.mark.parametrize(
+    ("status", "expected_ok"),
+    (
+        (200, True),
+        (201, True),
+        (301, True),
+        (400, False),
+        (403, False),
+        (500, False),
+    ),
+    ids=lambda tc: f"HTTP status code {tc[0]} is{' ' if tc[1] else ' not'} ok",
```

Comment:
@EvanKepner I cannot debug how to make it write in a browser. Could you please fix this locally?

Reply:
Yup, no problem - pushed a commit and it should pass CI now. `ids` is an optional arg, and with the structure of the tuple for parameterization I think pytest is unpacking as a list of lists which makes the lambda function difficult. I dropped `ids` entirely to use the default values and changed the first arg to be a single string. 

The other way around this would be to do something like:

```python
from typing import NamedTuple

class OKCases(NamedTuple):
    status: int
    expected_ok: boolean
    
    @property 
    def idfn(self):
        return f"{self.status} is {self.expected_ok}"
```

For the parameterized tuples to access as a single value but that feels excessive just for ids that already print the values.

Reply:
Well, the previous version of lambda relied on unpacking but still was breaking in the CI somehow. Thanks for the fix anyway

Reply:
As for the param names, I prefer them to be a tuple of strings for readability